### PR TITLE
[luci] Fuse sub with conv (for FusePreActivationBatchNormPass)

### PR DESCRIPTION
**Pull request by jinevening**
This fuses sub with conv for FusePreActivationBatchNormPass

ONE-DCO-1.0-Signed-off-by: Hyukjin Jeong <hj1.jeong@samsung.com>

---
Related to: #4773
Draft PR: #4769

### CODE REVIEW

**Review by seanshpark — APPROVED** [review]
LGTM

File: compiler/luci/pass/src/FusePreActivationBatchNormPass.cpp [hunk 1]
```diff
@@ -299,11 +299,81 @@ luci::CircleSub *insert_sub(luci::CircleNode *pred, luci::CircleConst *beta)
...
+  if (opcode == luci::CircleOpcode::ADD)
+  {
+    auto add = loco::must_cast<luci::CircleAdd *>(node);
+    return add;
+  }
+  else if (opcode == luci::CircleOpcode::MAX_POOL_2D)
+  {
+    auto succ = loco::succs(node);
+    if (succ.size() != 1)
+      return nullptr;
```
Comment (mhs4670go) [inline compiler/luci/pass/src/FusePreActivationBatchNormPass.cpp:314]: Just curiosity, is there `MAX_POOL_2D` that has two or more outputs?

Comment (jinevening) [reply compiler/luci/pass/src/FusePreActivationBatchNormPass.cpp:314]: `MAX_POOL_2D` always has one output (i.e., produces one output tensor). The code is checking the number of successors that use the output.

Comment (seanshpark) [reply compiler/luci/pass/src/FusePreActivationBatchNormPass.cpp:314]:
```
    auto succ = loco::succs(node);
    if (succ.size() != 1)
      return nullptr;
```
FYI, `succ` holds nodes that use `node` as input

**Comment by jinevening** [comment]
@mhs4670go PTAL

**Review by mhs4670go — APPROVED** [review]
LGTM

**Comment by jinevening** [comment]
@seanshpark  Can you merge this?

**Review by seanshpark — APPROVED** [review]
LGTM

**Pull request merged**